src/sinks/doris/client.rs

```diff
@@ -0,0 +1,437 @@
+            .await?;
+        let endpoint = request.uri().to_string();
+        let byte_size = payload.len();
+
+        let mut response = self.http_client.send(request).await?;
+        let mut status = response.status();
+
+        // Track protocol for metrics
+        let protocol = if endpoint.starts_with("https://") {
+            "https"
+        } else {
+            "http"
+        };
+
+        // Handle redirect loop
+        while (status == StatusCode::TEMPORARY_REDIRECT
+            || status == StatusCode::PERMANENT_REDIRECT
+            || status == StatusCode::FOUND)
+            && redirect_count < MAX_REDIRECTS
+        {
+            // Get redirect location
+            if let Some(location) = response.headers().get(http::header::LOCATION) {
+                if let Ok(location_str) = location.to_str() {
+                    debug!(
+                        message = "Following redirect.",
+                        status = %status,
+                        to = %location_str,
+                        redirect_count = redirect_count + 1
+                    );
+
+                    // Check for redirect loop
+                    if !visited_urls.insert(location_str.to_string()) {
+                        return Err(crate::Error::from("Detected redirect loop"));
+                    }
+
+                    // Build and send redirect request
+                    let redirect_req = self
+                        .build_request(&database, &table, payload_ref, Some(location_str))
+                        .await?;
+
+                    response = self.http_client.send(redirect_req).await?;
+                    status = response.status();
+
+                    // Increment redirect counter
+                    redirect_count += 1;
+
+                    debug!(
+                        message = "Received response after redirect.",
+                        new_status = %status,
+                        redirect_count = redirect_count
+                    );
+                } else {
+                    return Err(crate::Error::from(
+                        "Invalid Location header in redirect response",
+                    ));
+                }
+            } else {
+                return Err(crate::Error::from(
+                    "Missing Location header in redirect response",
+                ));
+            }
+        }
+
+        // Check if maximum redirects exceeded
+        if redirect_count >= MAX_REDIRECTS {
+            return Err(crate::Error::from(format!(
+                "Exceeded maximum number of redirects ({})",
+                MAX_REDIRECTS
+            )));
+        }
+
+        // Log endpoint bytes sent metric
+        emit!(EndpointBytesSent {
+            byte_size,
+            protocol,
+            endpoint: endpoint.as_str(),
+        });
+
+        // Extract response body
+        let (parts, body) = response.into_parts();
+        let body_bytes = hyper::body::to_bytes(body)
```

Comment:
If you merge in master I'm 99% sure this will warn now after a recent PR of mine. See changes made in #24351 for a guide on how to fix the warning